.eslintrc.json

```
@@ -4,7 +4,15 @@
   "extends": ["plugin:github/recommended", "plugin:github/typescript", "plugin:github/browser"],
   "rules": {
     "no-invalid-this": "off",
-    "@typescript-eslint/no-invalid-this": ["error"]
+    "@typescript-eslint/no-invalid-this": ["error"],
+    "import/extensions": [
+      "error",
+      "ignorePackages",
+      {
+        "ts": "never"
+      }
+    ],
```

By setting `.ts` to never, it does not _disallow_ the `.ts` extension, rather it says that if a file can be resolved to `.ts` then you can't use the `.ts` extension for the imports. In other words this is kind of the opposite of what we want. We _always_ want to supply an extension. If we remove this `ts` config and have a config like:

```suggestion
    "import/extensions": ["error", "ignorePackages"],
```

Then we _must_ supply an extension. If we supply a `.ts` extension, this lint rule will pass but `tsc` will fail telling us to not use the `.ts` extension.

Unfortunately these rules then give contradictory error messages, which is disappointing:

If I use no extension, then the lint rule tells me:

```
4:57  error Missing file extension "ts" for "./attr" import/extensions
```

If I fix it by adding a `.ts` extension, then `tsc` tells me:

```
src/controller.ts:4:57 - error TS2691: An import path cannot end with a '.ts' extension. Consider importing './attr' instead.
```

The only way to resolve it is to add a `.js` extension, at which point `tsc` and `eslint` are happy.